src/atoms/TextInput/TextInput.web.js

```diff
@@ -1,3 +1,508 @@
+  onChange,
+  onBlur,
+  helpText,
+  errorText,
+  variant,
+  prefix,
+  suffix,
+  disabled,
+  value,
+  iconLeft,
+  iconRight,
+  maxLength,
+  label,
+  testID,
+  labelPosition,
+  width,
+  type,
+  id,
+  name,
+  _isMultiline,
+}) => {
+  const theme = useContext(ThemeContext);
+  const inputRef = useRef();
+  const containerRef = useRef();
+  const [isFocused, setIsFocused] = useState(false);
+  const [input, setInput] = useState(value || '');
+  // Used for storing layout value of TextInput
+  const [layoutDimensions, setLayoutDimensions] = useState(null);
+  // Used to hide placeholder while label is inside the TextInput
+  const [isPlaceholderVisible, setIsPlaceholderVisible] = useState(isFocused);
+
+  const hasText = !isEmpty(input);
+  const hasAnimatedLabel = variant === 'outlined';
+
+  const placeholderTextColor = getPlaceholderTextColor({
+    theme,
+    disabled,
+    isPlaceholderVisible,
+    hasAnimatedLabel,
+  });
+
+  // Derive accessory conditions based on props
+  const { hasError, hasPrefix, hasSuffix, hasLeftIcon, hasRightIcon } = getAccessoryConfig({
+    errorText,
+    prefix,
+    suffix,
+    iconLeft,
+    iconRight,
+  });
+
+  if (labelPosition === 'left' && variant === 'outlined') {
+    // Outlined Text Input has only a top label
+    throw Error('Cannot have a left label on an outlined Text Input');
+  }
+
+  if (!isEmpty(prefix) && !isEmpty(iconLeft)) {
+    throw Error('Cannot have prefix and left icon together');
+  }
+
+  if (!isEmpty(suffix) && !isEmpty(iconRight)) {
+    throw Error('Cannot have suffix and right icon together');
+  }
+
+  const onFocus = useCallback(() => {
+    setIsFocused(true);
+    /* Wait for 90ms to show the placeholder since it takes 100ms for Label to animate from inside to top of the TextInput.
+       Otherwise they both overlap */
+    /* Don't have any delay if label is on left of TextInput */
+
+    setTimeout(() => {
+      setIsPlaceholderVisible(true);
+    }, 90);
+  }, [setIsFocused, setIsPlaceholderVisible]);
+
+  const onBlurText = useCallback(
+    (event) => {
+      setIsFocused(false);
+      setIsPlaceholderVisible(false);
+      const inputValue = event.target.value;
+      setInput(inputValue);
+      if (onBlur) {
+        onBlur(inputValue);
+      }
+    },
+    [setIsFocused, setIsPlaceholderVisible, setInput, onBlur],
+  );
+
+  const onChangeText = useCallback(
+    (event) => {
+      const inputValue = event.target.value;
+      if (inputValue.length > maxLength) {
+        return;
+      }
+      setInput(inputValue);
+      if (onChange) {
+        onChange(inputValue);
+      }
+    },
+    [maxLength, onChange, setInput],
+  );
+
+  const onKeyPress = useCallback(
+    (event) => {
+      if (type === 'number') {
+        const charCode = typeof event.which === 'number' ? event.which : event.keyCode;
+        const char = String.fromCharCode(charCode);
+        const isAllowed = /[0-9]/g.test(char) || char === '-' || char === '.';
```

Comment:
You can do check for `-` and `.` in the same regex itself. Also, add a small comment above that these are the allowed characters.